zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -778,6 +778,9 @@ public void shutdown() {
      */
     public synchronized void shutdown(boolean fullyShutDown) {
         if (!canShutdown()) {
+            if (fullyShutDown && zkDb != null) {
+                zkDb.clear();
```

Comment:
It's not obvious to me how we could end up in this code path - if we are here then the shutdown must have been called *and* that specific shut down must not be a full shutdown. Do we have a concrete example when we will hit this code path?

Reply:
In case the follower exited before finish syncing with leader, the ZK server will not start, canShutdown will return false, but we still need to clear the db here.